tests/phpunit/tests/test-switch-blog-rewrite-rules.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+if ( is_multisite() ) :
+
+	class Switch_Blog_Rewrite_Rules_Test extends PLL_Multisites_TestCase {
+		public function test_rewrite_rules_when_switching_blog() {
+			global $wp_rewrite;
+
+			$this->clean_up_filters();
+
+			$pll_admin = $this->get_pll_admin_env();
+			do_action_ref_array( 'pll_init', array( &$pll_admin ) );
+
+			/*
+			 * Test blog with Polylang activated and pretty permalink with language as directory.
+			 */
+			switch_to_blog( (int) self::$blog_with_pll_directory->blog_id );
+
+			$wp_rewrite->init();
+			$wp_rewrite->flush_rules();
+			$rules = $wp_rewrite->wp_rewrite_rules();
+
+			$this->assertNotEmpty( $rules );
+			$this->assertArrayNotHasKey( '(en)/?$', $rules );
+			$this->assertArrayHasKey( '(fr)/?$', $rules );
+			$this->assertArrayNotHasKey( '(fr)/(fr)/?$', $rules );
+
+			$languages = $pll_admin->model->get_languages_list();
+
+			$this->assertCount( 2, $languages );
+			$this->assertSame( 'en', $languages[0]->slug );
+			$this->assertSame( 'http://' . self::$blog_with_pll_directory->domain . self::$blog_with_pll_directory->path, $languages[0]->get_home_url() );
+			$this->assertSame( 'fr', $languages[1]->slug );
+			$this->assertSame( 'http://' . self::$blog_with_pll_directory->domain . self::$blog_with_pll_directory->path . 'fr/', $languages[1]->get_home_url() );
```

Comment:
If you are obliged to add the path in the url, then it means that there is no domain mapping. So I expect that it's useless to create sites on a specific domain and you can keep the domain param empty to keep the default one.

Reply:
Domains removed in [4b3bdc6](https://github.com/polylang/polylang/pull/1352/commits/4b3bdc6ecc7ee94cd8641d7270528c541266ed17).